packages/tocco-ui/src/EditableValue/typeEditors/MultiDateEdit.js

```diff
@@ -0,0 +1,165 @@
+import PropTypes from 'prop-types'
+import React from 'react'
+import moment from 'moment'
+import {injectIntl, intlShape} from 'react-intl'
+
+import DateAbstract from './DateAbstract'
+import {atMostOne, momentJStoToFlatpickrFormat, toLocalDateString} from '../utils'
+
+class MultiDateEdit extends React.Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      showCalendar: true,
+      showDoubleCalendar: false,
+      counter: 0,
+      label: this.RANGE_CALENDAR_LABEL
+    }
+    this.onChangeObject = {
```

Comment:
why don't you handle the values in the state? I guess that's what React State is for? :)

Reply:
When switching `setState()` wasn't the right fit because it's async.
From https://reactjs.org/docs/faq-state.html#when-is-setstate-asynchronous:

> Currently, setState is asynchronous inside event handlers.